pkg/cloudprovider/aws/aws.go

```diff
@@ -73,12 +107,65 @@ func (f *ec2InstanceFilter) Matches(instance ec2.Instance) bool {
 }
 
 // goamzEC2 is an implementation of the EC2 interface, backed by goamz
-type GoamzEC2 struct {
-	ec2 *ec2.EC2
+type goamzEC2 struct {
+	auth aws.Auth
+	ec2  *ec2.EC2
+
+	mutex      sync.Mutex
+	elbClients map[string]*elb.ELB
+}
+
+func newGoamzEC2(auth aws.Auth, regionName string) (*goamzEC2, error) {
+	region, ok := aws.Regions[regionName]
+	if !ok {
+		return nil, fmt.Errorf("not a valid AWS region: %s", regionName)
+	}
+
+	self := &goamzEC2{}
+	self.ec2 = ec2.New(auth, region)
+	self.auth = auth
+	self.elbClients = make(map[string]*elb.ELB)
+	return self, nil
+}
+
+// Find the kubernetes vpc
+func (self *goamzEC2) ListVpcs(filterName string) ([]ec2.VPC, error) {
+	client := self.ec2
+
+	// TODO: How do we want to identify our VPC?
```

Comment:
This is all #6006 again.  I think I want to have a per-k8s-cluster id, and then I will add a filter for that.  The name is shown in the AWS console, so is for humans, and I could imagine people wanting to change the name.

The idea I have is to only filter on tags that have a "k8s:" prefix; changing a tag with a k8s prefix will break things, but changing the Name tag should be fair game.